tests/unit/s2n_tls13_compute_shared_secret_test.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include "testlib/s2n_testlib.h"
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#include <s2n.h>
+
+/* Just to get access to the static functions / variables we need to test */
+#include "tls/s2n_tls13_handshake.c"
+
+
+int main(int argc, char **argv) {
+
+    BEGIN_TEST();
+
+    struct s2n_connection *client_conn;
+
+    /* This file ensures the function s2n_tls_compute_shared_secret is correctly error-checking for missing data
+     * in the server hello. */
+
+    {
+        /* This test ensures that if the server did not send a keyshare extension in the server hello function,
+         * a null pointer error is correctly thrown. */
+
+
+        EXPECT_NOT_NULL(client_conn = s2n_connection_new(S2N_CLIENT));
+
+        client_conn->actual_protocol_version = S2N_TLS13;
+
+        /* Select curve and generate key for client */
+        client_conn->secure.client_ecc_evp_params[0].negotiated_curve = client_conn->config->ecc_preferences->ecc_curves[0];
+        EXPECT_SUCCESS(s2n_ecc_evp_generate_ephemeral_key(&client_conn->secure.client_ecc_evp_params[0]));
+
+        DEFER_CLEANUP(struct s2n_blob client_shared_secret = {0}, s2n_free);
+        /* Compute fails because server's curve and public key are missing. */
+        EXPECT_FAILURE_WITH_ERRNO(s2n_tls13_compute_shared_secret(client_conn, &client_shared_secret), S2N_ERR_NULL);
```

Comment:
Can we add a test case that explicitly checks for when `client_conn->secure.server_ecc_evp_params.negotiated_curve = NULL`. In this case as server_ecc_evp_params is not set, the error is returned from this check `notnull_check(server_key);` on tls/s2n_tls13_handshake.c  Line 60.